runtime/version/fork.go

```diff
@@ -1,28 +1,58 @@
 package version
 
+import "github.com/pkg/errors"
+
 const (
 	Phase0 = iota
 	Altair
 	Bellatrix
 	Capella
 )
 
+var versionToString = map[int]string{
+	Phase0:    "phase0",
+	Altair:    "altair",
+	Bellatrix: "bellatrix",
+	Capella:   "cappella",
+}
+
+// stringToVersion and allVersions are populated in init()
+var stringToVersion = map[string]int{}
+var allVersions = []int{}
+
+// ErrUnrecognizedVersionName means a string does not match the list of canonical version names.
+var ErrUnrecognizedVersionName = errors.New("version name doesn't map to a known value in the enum")
+
+// FromString translates a canonical version name to the version number.
+func FromString(name string) (int, error) {
+	v, ok := stringToVersion[name]
+	if !ok {
+		return 0, errors.Wrap(ErrUnrecognizedVersionName, name)
+	}
+	return v, nil
+}
+
+// String returns the canonical string form of a version.
+// Unrecognized versions won't generate an error and are represented by the string "unknown version".
 func String(version int) string {
-	switch version {
-	case Phase0:
-		return "phase0"
-	case Altair:
-		return "altair"
-	case Bellatrix:
-		return "bellatrix"
-	case Capella:
-		return "capella"
-	default:
```

Comment:
I changed this so that the `versionToString` map acts as a source of truth - when we add something to it the `stringToVersion` map and `allVersions` get updated automatically.